### PR TITLE
Site settings

### PR DESCRIPTION
**Pull request by JSey**
implements the feature requests from here: https://meta.discourse.org/t/babble-a-chat-plugin/31753/56?u=jsey

- Babble icon
- custom placeholder
- en/disable plugin completely

### CODE REVIEW

File: assets/javascripts/discourse/templates/components/babble-icon.hbs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-                     icon="bullhorn"
-                     toggleVisible=babbleVisible
-                     loginAction="showLogin"
-                     title="babble.title"}}
-  {{/header-dropdown}}
-  {{#if unreadCount}}
-    <a href class='badge-notification unread-notifications'>{{unreadCount}}</a>
-  {{/if}}
-{{/unless}}
+{{#if Discourse.SiteSettings.babble_enabled}}
```
Comment (gdpelican) [inline assets/javascripts/discourse/templates/components/babble-icon.hbs]: I'd love to move this into a single sanity check with `noTopicAvailable`

What about
```
{{#if babbleEnabled}}
  ...
{{/if}}
```

```
Discourse.Babble.refresh = function(data) {
  self.set('babbleEnabled', Discourse.SiteSettings.babble_enabled && data.id
  if (!self.get('babbleEnabled') { return }
}
```

?

File: assets/javascripts/discourse/components/babble-composer.js.es6 [hunk 2]
```diff
@@ -4,7 +4,11 @@ export default Ember.Component.extend({
   classNames: ['babble-post-composer'],
 
   _init: function() {
-    this.set('placeholder', I18n.t('babble.placeholder'))
+    if (Discourse.SiteSettings.babble_placeholder) {
+      this.set('placeholder', Discourse.SiteSettings.babble_placeholder)
```
Comment (gdpelican) [inline assets/javascripts/discourse/components/babble-composer.js.es6]: I reckon just

```
this.set('placeholder', Discourse.SiteSettings.babble_placeholder || I18n.t('babble.placeholder'))
```

should do the trick here.

File: assets/javascripts/discourse/templates/components/babble-icon.hbs (hunk 1 above)
Comment (JSey) [inline assets/javascripts/discourse/templates/components/babble-icon.hbs]: Great. Will do.

File: assets/javascripts/discourse/components/babble-composer.js.es6 (hunk 2 above)
Comment (JSey) [inline assets/javascripts/discourse/components/babble-composer.js.es6]: Oh, sure. It's too late here to write simple code. :)

**Comment by JSey** [comment]
Done... :)

**Comment by gdpelican** [comment]
Great, thanks! I merged this manually to get all the extra commits out and bump the version, but it's in master now.

**Comment by JSey** [comment]
Thanks!

File: assets/javascripts/discourse/templates/components/babble-icon.hbs [hunk 3]
```diff
@@ -1,6 +1,6 @@
-{{#unless noTopicAvailable}}
+{{#if babbleEnabled}}
   {{#header-dropdown iconId="babble-notifications"
-                     icon="bullhorn"
+                     icon=Discourse.SiteSettings.babble_icon
```
Comment (gdpelican) [inline assets/javascripts/discourse/templates/components/babble-icon.hbs]: Hey @JSey, there's a deprecation around using Discourse.SiteSettings in the view, can we do something like
```
icon=babbleIcon
```
and in the controller
```
self.set('babbleIcon', Discourse.SiteSettings.babble_icon)
```
?

**Pull request closed without merging**